cluster-provision/k8s/1.30/k8s_provision.sh

```diff
@@ -148,9 +148,24 @@ patch $cni_manifest_ipv6 $cni_ipv6_diff
 
 cp /tmp/local-volume.yaml /provision/local-volume.yaml
 
-# TODO use config file! this is deprecated
+# Create drop-in config files for kubelet
+# https://kubernetes.io/docs/tasks/administer-cluster/kubelet-config-file/#kubelet-conf-d
+kubelet_conf_d="/etc/kubernetes/kubelet.conf.d"
+mkdir -m 644 $kubelet_conf_d
+
+# Set our custom initializations to kubelet
+kubevirt_kubelet_conf="$kubelet_conf_d/50-kubevirt.conf"
+cat <<EOF >$kubevirt_kubelet_conf
+apiVersion: kubelet.config.k8s.io/v1beta1
+kind: KubeletConfiguration
+cgroupDriver: systemd
+failSwapOn: false
+kubeletCgroups: /systemd/system.slice
+EOF
+
+# Set only command line options not supported by config
```

Comment:
Looks great to me!

Do you think it's possible to somehow let the user provide an alternative `KubeletConfiguration` that will override this? I think it would give the user full control which could be very beneficial when doing complex configuration. In any case it should not block this PR, just an idea :)